lib/hasher/src/CRC64NVMEStrategy.cpp

```diff
@@ -0,0 +1,81 @@
+#include "irods/CRC64NVMEStrategy.hpp"
+
+#include "irods/rodsErrorTable.h"
+#include "irods/checksum.h"
+#include "irods/base64.hpp"
+
+#include <cstring>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+#include <fmt/format.h>
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/crc.hpp>
+
+namespace irods
+{
+
+    const std::string CRC64NVME_NAME("crc64nvme");
+
+    auto CRC64NVMEStrategy::init(boost::any& _context) const -> error
+    {
+        constexpr uint64_t crc_polynomial = 0xAD93D23594C93659;
+        constexpr uint64_t crc_initial_remainder = 0xFFFFFFFFFFFFFFFF;
+        constexpr uint64_t crc_final_xor = 0xFFFFFFFFFFFFFFFF;
+        constexpr bool crc_reflect_input = true;
+        constexpr bool crc_reflect_output = true;
+
+        // Create the CRC calculator object
+        boost::crc_basic<crc_bits>* context = new boost::crc_basic<crc_bits>(
+            crc_polynomial, crc_initial_remainder, crc_final_xor, crc_reflect_input, crc_reflect_output);
```

Comment:
Is there a reason why the context is heap allocated?

Reply:
This context needs to survive longer than this function call.

This is similar to the others.
https://github.com/irods/irods/blob/b8781c11aef62b61b3ff3e29d17b52bb105fe372/lib/hasher/src/SHA256Strategy.cpp#L30
https://github.com/irods/irods/blob/b8781c11aef62b61b3ff3e29d17b52bb105fe372/lib/hasher/src/SHA256Strategy.cpp#L86

Reply:
Sure. If the `boost:crc_basic<crc_bits>` type supports copy/move semantics, then you can avoid the heap allocation. That would also remove the need for anyone to protect against memory leaks.

With that said, code which relies on these hashers might assume data generated by them are heap allocated. If that's true, then we should continue with the existing approach.

I'm fine with your solution overall. Feel free to mark this as resolved.